src/grisbi_osx.m

```diff
@@ -108,36 +117,68 @@ static void my_setenv(const gchar *key, const gchar *value) {
 
     // XDG
     // https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
-    my_setenv("XDG_DATA_HOME",   g_build_filename(app_support_dir, "share", NULL));
+    gchar *support_share_dir = g_build_filename(app_support_dir, "share", NULL);
+    my_setenv("XDG_DATA_HOME",   support_share_dir);
+    g_free(support_share_dir);
+
     my_setenv("XDG_DATA_DIRS",   bundle_resources_share_dir);
-    my_setenv("XDG_CONFIG_HOME", g_build_filename(app_support_dir, "config", NULL));
-    my_setenv("XDG_CONFIG_DIRS", g_build_filename(bundle_resources_etc_dir, "xdg", NULL));
-    my_setenv("XDG_CACHE_HOME",  g_build_filename(app_support_dir, "cache", NULL));
+
+    gchar *support_config_dir = g_build_filename(app_support_dir, "config", NULL);
+    my_setenv("XDG_CONFIG_HOME", support_config_dir);
+    g_free(support_config_dir);
```

Comment:
I suggest to create a macro that would do something like:
`SETENV("XDG_CONFIG_HOME", g_build_filename(app_support_dir, "config", NULL));`

With:
`#define SETENV(variable, value) do {gchar *tmp = value; my_setenv(variable, tmp); g_free(tmp);} while(0)`
The idea is to remove code duplication and simplify code reading/understanding.

The code of `my_setenv()` could also be moved inside the macro.

Reply:
nope, some values may be commo for few variables.